app/models/service_template.rb

```diff
@@ -434,6 +444,24 @@ def order(user_or_id, options = nil, request_options = nil, schedule_time = nil)
     end
   end
 
+  def update_schedule(schedule_time)
+    if schedule_time
+      require 'time'
+      time = Time.parse(schedule_time).utc
+
+      schedule = miq_schedule
+      raise _('service_template is not currently scheduled') unless schedule
+      schedule.run_at = {
+        :interval   => {:unit => "once"},
+        :start_time => time,
+        :tz         => "UTC",
+      }
+      schedule.save!
+    else
+      miq_schedule&.destroy
```

Comment:
I find it a bit odd that we destroy an associated schedule here in `update_schedule` method but cannot create a schedule if the `schedule_time` is provided, see the raise on line 453.  I feel like this method should only update an existing schedule, not delete.  If we can't update it, we should raise.  If they want to delete it, have another method for deleting the schedule.

```ruby
def update_schedule(schedule_time)
  raise ... unless schedule_time
  ...
  raise ... unless schedule
  ...save!
  true
end
```